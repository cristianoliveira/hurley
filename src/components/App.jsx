import React, { Component } from 'react';
import { Link } from 'react-router';
import logo from './../images/logo.svg';

import './../styles/App.css';

import feedcastApi from './../scripts/feedcastApi'
import Search from './Search.jsx'
import PlayerFooter from './PlayerFooter.jsx'

class App extends Component {

  render() {

    return (
      <div className="feedcast">
        <div className="feedcast__header">
          <Link to="/" className="feedcast__logo-wrapper">
            <img src={logo} className="feedcast__logo" alt="logo" />
          </Link>
          <Search/>
        </div>
        <div className="feedcast__navbar">
          <Link activeClassName="active" to="channels"><i className="fa fa-rss"></i> Canais</Link>
          <Link activeClassName="active" to="lastEpisodes"><i className="fa fa-history"></i> Últimos Episódios</Link>
          <Link activeClassName="active" to="categoriesList"><i className="fa fa-th-large"></i> Categorias</Link>
        </div>
        <div className="feedcast__container">
          {this.props.children}
        </div>
        <PlayerFooter />
      </div>
    );
  }
}

export default App;
