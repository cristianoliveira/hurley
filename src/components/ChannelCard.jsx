import React, { Component } from 'react'
import { Link } from 'react-router'

import helpers from './../scripts/helpers'

export default class ChannelCard extends Component {

  constructor(props) {
    super(props);
    this.state = {
      imageStatus : false,
      image_url : ''
    }
  }


  componentDidMount() {
    this.checkImage(this.props)
  }


  componentDidUpdate(prevProps) {
    if(this.props.data.image_url !==
        prevProps.data.image_url){
      this.checkImage(this.props)
    }
  }


  checkImage(props){
    let c = props.data
    helpers
      .testImage(c.image_url, 5000)
      .then(e => {
        this.setState({
          imageStatus: true })
      })
      .catch(e => {
        this.setState({
          imageStatus: false })
      })
  }


  thumbnail({image_url}){
    return this.state.imageStatus ? (
      <img
        src={image_url}
        className="feedcast__channel-img"
        onError={()=>this.setState({imageStatus: false})}
      />
    ) : (
      <div className="feedcast__channel-thumbnail">
        <i className="fa fa-podcast"></i>
      </div>
    )
  }

  render () {
    let c = this.props.data
    let thumb = this.thumbnail(c)

    return (
      <Link key={c.uuid} to={`/channel/${c.uuid}`}>
        <div className="feedcast__channel" id={c.uuid}>
          <div className="feedcast__channel-title">
            <h3>{c.title}</h3>
          </div>
          {thumb}
          <div className="feedcast__channel-overlay"></div>
        </div>
      </Link>
    )
  }

}
