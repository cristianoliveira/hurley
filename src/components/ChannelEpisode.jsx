import React, { Component } from 'react'
import helpers from './../scripts/helpers'

class ChannelEpisode extends Component {

  constructor(){
    super();

    this.state = {
      seeMore: false
    }
  }

  render(){

    const { title, summary, audio, description } = this.props.episode
    const descText = summary && summary.length > 0 ? summary : description;

    const summaryEl = descText && descText.length > 0 ? (
      <p className={`feedcast__channelEpisode-summary
        feedcast__channelEpisode-summary--${ this.state.seeMore
          ? 'see-more': 'see-less'}`} >
        {descText}
      </p>
    ):'';

    const btnSeeMore = descText && descText.length > 0 ? (
      <button
        className="feedcast__channelEpisode-btn-see-more"
        onClick={()=> this.setState(
          { seeMore: !this.state.seeMore})}>
        {this.state.seeMore ? 'Ver Menos' : 'Ver Mais'}
      </button>
    ) : '';

    return (
      <div className="feedcast__channelEpisode">
        <div className="feedcast__channelEpisode-button-wrapper">
          <button className="feedcast__channelEpisode-play-button">
            <i className="fa fa-play"></i>
          </button>
        </div>
        <div className="feedcast__channelEpisode-info-wrapper">
          <h4>{title}</h4>
          {summaryEl}
          {btnSeeMore}
          <span>{helpers.secondsToHms(audio.duration)}</span>
        </div>
      </div>
    )
  }
}


export default ChannelEpisode
