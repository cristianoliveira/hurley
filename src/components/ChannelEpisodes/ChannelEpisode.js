import React, { Component } from 'react'
import { connect } from 'react-redux';

import helpers from 'app/scripts/helpers'
import { playEpisode } from 'app/actions/player';

class ChannelEpisode extends Component {

  constructor(){
    super();

    let { lc } = helpers.localize(this)

    this.state = {
      seeMore: false,
      lc
    }
  }

  playEpisode(episode){
    this.props.dispatch(playEpisode(episode));
  }

  render(){
    const { lc } = this.state
    const { title, summary, audio, description } = this.props.episode
    const descText = summary && summary.length > 0 ? summary : description;

    const summaryEl = descText && descText.length > 0 ? (
      <p className={`feedcast__channelEpisode-summary
        feedcast__channelEpisode-summary--${ this.state.seeMore
          ? 'see-more': 'see-less'}`} >
        {descText}
      </p>
    ):'';

    const btnSeeMore = descText && descText.length > 0 ? (
      <button
        className="feedcast__channelEpisode-btn-see-more"
        onClick={()=> this.setState(
          { seeMore: !this.state.seeMore})}>
        {this.state.seeMore ? lc.seeLess : lc.seeMore}
      </button>
    ) : '';

    return (
      <div className="feedcast__channelEpisode">
        <div className="feedcast__channelEpisode-button-wrapper">
          <button
            onClick={() => this.playEpisode(this.props.episode)}
            className="feedcast__channelEpisode-play-button">
            <i className="fa fa-play"></i>
          </button>
        </div>
        <div className="feedcast__channelEpisode-info-wrapper">
          <h4>{title}</h4>
          {summaryEl}
          {btnSeeMore}
          <span>{helpers.secondsToHms(audio.duration)}</span>
        </div>
      </div>
    )
  }
}


export default connect()(ChannelEpisode)
