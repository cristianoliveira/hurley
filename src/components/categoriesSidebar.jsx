import React from 'react';
import { Link } from 'react-router';


import feedcastApi from './../scripts/feedcastApi'


export default class CategoriesSidebar extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      categories : []
    }
  }





  componentDidMount() {
    feedcastApi
    .getCategories()
    .then( data => {
        this.setState({categories : data.categories })
    })
  }




  render(){
    return (
      <div className="feedcast__categories-sidebar">
        { this.state.categories.map((c, i)=>(
          <Link key={i} to={`/category/${c.slug}`}>
            <i className={`fa fa-${c.icon}`}></i> {c.title}
          </Link>)) }
      </div>
    )
  }
}
