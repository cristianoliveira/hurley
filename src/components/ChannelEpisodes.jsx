import React, { Component } from 'react'
import { Link } from 'react-router';
import feedcastApi from './../scripts/feedcastApi'

import ChannelEpisode from './ChannelEpisode.jsx'

class ChannelEpisodes extends Component {

  constructor(){
    super();

    this.state = {
      episodes : [],
      per_page: 10,
      page : 1,
      uuid: null,
      total: 50
    }
  }

  componentWillReceiveProps(nextProps){
    const {data} = nextProps
    let page = typeof data.page === 'string' ? parseInt(data.page) : 1
    feedcastApi
      .getChannelEpisodes(data)
      .then( res => {
        this.setState({
          episodes : res.episodes,
          page,
          uuid : data.uuid,
          total: res.total
        })
      })
  }


  listEpisodes(){
    let list = this.state.episodes.map( (i, n) =>
      <ChannelEpisode key={n} episode={i}/> )
    return this.state.episodes.length > 0 ? list : '';
  }



    navigation(){
    //TODO: Abstract this method to a component
    //used in other places to
    let { episodes, total, per_page, page} = this.state;
    let totalPages = parseInt(total / per_page);

    if(total % per_page > 0)
      totalPages++;

    let pages = []

    pages[1] = 1
    pages[page - 1] = page - 1
    pages[page] = page
    pages[page + 1] = page + 1
    pages[totalPages] = totalPages

    let buttons = pages
                    .filter(i => i > 0 && i <= totalPages)
                    .map(i=>(
        <Link
          key={i}
          to={`channel/${this.state.uuid}/${i}`}
          className={i == page ? 'active':''}>
          <button>{i}</button>
        </Link>
      ))

    return episodes.length > 0 && totalPages > 0 ? (
      <div className="feedcast__channel-navigation
      feedcast__channel-navigation--episodes">
        <Link to={`channel/${this.state.uuid}/${
          this.state.page > 1?
          (this.state.page - 1) : 1}`} >
          <button>
            <i className="fa fa-angle-double-left"></i>
          </button>
        </Link>
        {buttons}
        <Link to={`channel/${this.state.uuid}/${
          this.state.page < totalPages?
          (this.state.page + 1) : totalPages}`}>
          <button>
            <i className="fa fa-angle-double-right"></i>
          </button>
        </Link>
      </div>
    ) : ''
  }



  render(){
    const list = this.listEpisodes();
    const navButtons = this.navigation();

    return (
      <div className="feedcast__channelEpisodes">
        <h1>Lista de Epis√≥dios</h1>
        {list}
        {navButtons}
      </div>
    )
  }
}


export default ChannelEpisodes
