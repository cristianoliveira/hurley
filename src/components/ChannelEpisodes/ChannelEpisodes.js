import React, { Component } from 'react'
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import feedcastApi from 'feedcast-client';

import helpers from 'app/scripts/helpers'
import ChannelEpisode from './ChannelEpisode'
import Pagination from 'app/components/Pagination'

class ChannelEpisodes extends Component {
  static propTypes = {
    episodes: PropTypes.array.isRequired,
    channel: PropTypes.object.isRequired,
  }

  static defaultProps = {
    episodes: [],
    channel: {},
  }

  listEpisodes(){
    let list = this.props.episodes.map( (i, n) =>
      <ChannelEpisode key={n} episode={i}/> )
    return this.props.episodes.length > 0 ? list : '';
  }


  render(){
    const list = this.listEpisodes();


    return (
      <div className="feedcast__channelEpisodes">
        <h1>Episodes</h1>
        {list}
        <Pagination
          url={`/channel/${this.props.channel.uuid}/`}
          page={this.props.page}
          per_page={this.props.per_page}
          total={this.props.total}
          theme="white" />
      </div>
    )
  }
}


export default ChannelEpisodes
