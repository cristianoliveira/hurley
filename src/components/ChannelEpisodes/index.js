import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { requestEpisodesForChannel } from 'app/actions/episodes';
import FeedcastLoader from 'app/components/FeedcastLoader';
import ChannelEpisodes from './ChannelEpisodes';

class ChannelEpisodesContainer extends Component {

  static propTypes = {
    page: PropTypes.number,
    per_page: PropTypes.number,
    episode: PropTypes.object,
    episodes: PropTypes.array,
    isFetching: PropTypes.bool,
  }

  static defaultProps = {
    episode: {},
    episodes: [],
    isFetching: true,
    page: 0,
    per_page: 20,
  }

  componentDidMount() {
    const { page = '1' } = this.props.params;
    if (page != this.props.page)
      this.props.dispatch(requestEpisodesForChannel({
        uuid: this.props.channel.uuid,
        page: parseInt(page),
      }));
  }

  render() {
    if (this.props.isFetching) {
      return <FeedcastLoader />;
    }

    return <ChannelEpisodes {...this.props} />;
  }
}

const mapStateToProps = (state) => state.episodes;

export { ChannelEpisodes };
export default connect(mapStateToProps)(ChannelEpisodesContainer);
