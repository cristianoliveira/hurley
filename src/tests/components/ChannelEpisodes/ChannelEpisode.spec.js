import React from 'react';
import renderer from 'react-test-renderer';
import { withStore } from 'app/tests/helpers';
import { MemoryRouter } from 'react-router-dom'

import { ChannelEpisodes } from 'app/components/ChannelEpisodes';

describe('ChannelEpisodes', () => {
  describe('when has no data', () => {
    it("renders the no episode", () => {
      const component = renderer.create(
        <MemoryRouter initialEntries={[ `/pensando-rpg/episodes/1` ]}>
          <ChannelEpisodes />
        </MemoryRouter>
      );
      expect(component).toMatchSnapshot();
    })
  });

  describe('when has a list of episodes', () => {
    const state = {
      episode: {
        uuid: "aadwdadadwawdknaw",
        slug: "pensando-rpg-number-081-entenda-a-figura-do-bruxo-warlock",
        title: "Pensando RPG #081 - Entenda a figura do Bruxo (Warlock)",
        summary: "Neste episódio, falo sobre a figura do Bruxo e a sua respectiva classe em Dungeons & Dragons. Aqui, diferencio-o de outras classes mágicas e dou algumas ideias de jogabilidade e interpretação.",
        description: "Neste episódio, falo sobre a figura do Bruxo e a sua respectiva classe em Dungeons & Dragons. Aqui, diferencio-o de outras classes mágicas e dou algumas ideias de jogabilidade e interpretação.",
        published_at: "2017-08-10T21:58:31+00:00",
        audio: {
          url: "http://feeds.soundcloud.com/stream/337354860-user-142845446-pensando-rpg-081-entenda-a-figura-do-bruxo-warlock.mp3",
          size: 12170733,
          duration: 760.633469,
          codec: "mp3",
          bitrate: 128000,
          sample_rate: 44100,
          status: "analysed",
          analysed_at: "2017-08-10T22:17:19+00:00",
          error_message: "",
          error_count: 0
        },
      },
      channel: {
          uuid: "asdasdadasda",
          slug: "pensando-rpg",
          title: "Pensando RPG",
          description: "",
          feed_url: "http://feeds.soundcloud.com/users/soundcloud:users:257525576/sounds.rss",
          image_url: "https://img.feedcast.io/http%3A%2F%2Fi.imgur.com%2FmYRfJk7.png/300/300",
          listed: true,
          synchronization_status: "success",
          synchronization_status_message: "",
          channels: [],
        }
      ,
      page: 1,
      perPage: 130,
      total: 1000,
    }

    it('render the category', () => {
      const component = renderer.create(withStore(
        <MemoryRouter initialEntries={[ `/pensando-rpg/episodes/1` ]}>
          <ChannelEpisodes {...state} />
        </MemoryRouter>
      ));
      expect(component).toMatchSnapshot();
    });
  });
})
