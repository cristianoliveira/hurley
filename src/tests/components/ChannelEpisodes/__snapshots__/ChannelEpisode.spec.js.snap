// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelEpisodes when has a list of episodes render the category 1`] = `
<div
  className="feedcast__channelEpisodes"
>
  <div
    className="feedcast__channelEpisode"
  >
    <div
      className="feedcast__channelEpisode-button-wrapper"
    >
      <button
        className="feedcast__channelEpisode-play-button"
        onClick={[Function]}
      >
        <i
          className="fa fa-play"
        />
      </button>
    </div>
    <div
      className="feedcast__channelEpisode-info-wrapper"
    >
      <h4>
        Pensando RPG #081 - Entenda a figura do Bruxo (Warlock)
      </h4>
      <button>
        <i
          className="fa fa-ellipsis-h"
        />
      </button>
      <span>
        00:12:40
      </span>
    </div>
  </div>
  <div
    className="feedcast__pagination feedcast__pagination--white"
  >
    
  </div>
</div>
`;

exports[`ChannelEpisodes when has no data renders the no episode 1`] = `
<div
  className="feedcast__channelEpisodes"
>
  <div
    className="feedcast__pagination feedcast__pagination--white"
  >
    
  </div>
</div>
`;
